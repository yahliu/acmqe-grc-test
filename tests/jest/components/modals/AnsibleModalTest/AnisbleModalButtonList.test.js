/* Copyright (c) 2020 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'

import { buildModalButtonList } from '../../../../../src-web/components/modals/AnsibleAutomationModal/AnisbleModalButtonList'

describe('render action buttons on ansible modal', () => {
  it('render no action button for ansible history page', () => {
    // activeItem === 1 means ansible history page
    const component = buildModalButtonList({
      activeItem: 1, opInstalled: true,
      policyAutoName: 'test-policy-3-1624936287-policy-automation', locale: 'us',
      handleSubmitClick:jest.fn(), handleCloseClick:jest.fn(), handleOpenDelModal:jest.fn()
    })
    expect(component).toMatchSnapshot()
  })

  it('render disabled action button when ansible operator isn not installed but there is automation', () => {
    const component = buildModalButtonList({
      activeItem: 0, opInstalled: false,
      policyAutoName: 'test-policy-3-1624936287-policy-automation', locale: 'us',
      handleSubmitClick:jest.fn(), handleCloseClick:jest.fn(), handleOpenDelModal:jest.fn()
    })
    expect(component).toMatchSnapshot()
  })

  it('render two action buttons without delete button when no policy automation', () => {
    const component = buildModalButtonList({
      activeItem: 0, opInstalled: true,
      policyAutoName: '', locale: 'us',
      handleSubmitClick:jest.fn(), handleCloseClick:jest.fn(), handleOpenDelModal:jest.fn()
    })
    expect(component).toMatchSnapshot()
  })

  it('render three action buttons with delete button', () => {
    const component = buildModalButtonList({
      activeItem: 0, opInstalled: true,
      policyAutoName: 'test-policy-3-1624936287-policy-automation', locale: 'us',
      handleSubmitClick:jest.fn(), handleCloseClick:jest.fn(), handleOpenDelModal:jest.fn()
    })
    expect(component).toMatchSnapshot()
  })

  it('render three action buttons with disabled delete button with editOnly', () => {
    const component = buildModalButtonList({
      onlyEdit: true, activeItem: 0, opInstalled: true,
      policyAutoName: 'test-policy-3-1624936287-policy-automation', locale: 'us',
      handleSubmitClick:jest.fn(), handleCloseClick:jest.fn(), handleOpenDelModal:jest.fn()
    })
    expect(component).toMatchSnapshot()
  })
})
