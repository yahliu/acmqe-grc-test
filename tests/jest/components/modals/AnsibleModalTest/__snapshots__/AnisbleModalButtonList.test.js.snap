// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render action buttons on ansible modal render disabled action button when ansible operator isn not installed but there is automation 1`] = `
Array [
  <AcmButton
    isDisabled={true}
    onClick={[MockFunction]}
    variant="primary"
  >
    Save
  </AcmButton>,
  <AcmButton
    onClick={[MockFunction]}
    variant="secondary"
  >
    Cancel
  </AcmButton>,
  <AcmButton
    onClick={[MockFunction]}
    variant="danger"
  >
    Delete
  </AcmButton>,
]
`;

exports[`render action buttons on ansible modal render no action button for ansible history page 1`] = `Array []`;

exports[`render action buttons on ansible modal render three action buttons with delete button 1`] = `
Array [
  <AcmButton
    isDisabled={false}
    onClick={[MockFunction]}
    variant="primary"
  >
    Save
  </AcmButton>,
  <AcmButton
    onClick={[MockFunction]}
    variant="secondary"
  >
    Cancel
  </AcmButton>,
  <AcmButton
    onClick={[MockFunction]}
    variant="danger"
  >
    Delete
  </AcmButton>,
]
`;

exports[`render action buttons on ansible modal render three action buttons with disabled delete button with editOnly 1`] = `
Array [
  <AcmButton
    isDisabled={false}
    onClick={[MockFunction]}
    variant="primary"
  >
    Save
  </AcmButton>,
  <AcmButton
    onClick={[MockFunction]}
    variant="secondary"
  >
    Cancel
  </AcmButton>,
  <mockTooltip
    content="You are not authorized to complete this action. See your cluster administrator for role-based access information."
    zIndex={10001}
  >
    <div
      className="disable-tooltip"
    >
      <AcmButton
        isDisabled={true}
        variant="danger"
      >
        Delete
      </AcmButton>
    </div>
  </mockTooltip>,
]
`;

exports[`render action buttons on ansible modal render two action buttons without delete button when no policy automation 1`] = `
Array [
  <AcmButton
    isDisabled={false}
    onClick={[MockFunction]}
    variant="primary"
  >
    Save
  </AcmButton>,
  <AcmButton
    onClick={[MockFunction]}
    variant="secondary"
  >
    Cancel
  </AcmButton>,
]
`;
